### PR TITLE
[Enhance] WandbVisBackend supports updating config

### PR DESCRIPTION
**Pull request by zgzhengSEU**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

support wandb resume

## Modification

Please briefly describe what modification is made in this PR.

support wandb resume

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMCls.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: mmengine/visualization/vis_backend.py [hunk 1]
```diff
@@ -417,7 +417,12 @@ def add_config(self, config: Config, **kwargs) -> None:
         Args:
             config (Config): The Config object
         """
-        self._wandb.config.update(dict(config))
+        if 'allow_val_change' in self._init_kwargs:
+            self.allow_val_change = self._init_kwargs['allow_val_change']
+        else: 
+            self.allow_val_change = False
+        self._wandb.config.update(
+            dict(config), allow_val_change=self.allow_val_change)
```
Comment (zhouzaida) [inline mmengine/visualization/vis_backend.py:425]: If `allow_val_change` exists in `self._init_kwargs`, it should be passed in `wandb` when calling `wandb.init(**self._init_kwargs)` in `_init_env` method.

Comment (zgzhengSEU) [reply mmengine/visualization/vis_backend.py:425]: image
I have tried to set [resume=True/"allow", id="xxx", allow_val_change=True] in wandb.init, but still report this error。
But After I set the allow_val_change=True in wandb.config.update(), it stopped prompting this error and successfully ran resume

Comment (zhouzaida) [reply mmengine/visualization/vis_backend.py:425]: Hi @zgzhengSEU , thanks for your contribution. Could you also open an issue in wandb to report this issue?

Comment (zhouzaida) [inline mmengine/visualization/vis_backend.py:425]:
```suggestion
        allow_val_change = self._init_kwargs.get('allow_val_change', False)
        self._wandb.config.update(
            dict(config), allow_val_change=allow_val_change)
```

**Review by zhouzaida — APPROVED** [review]

**Pull request merged**